packages/icon-button/src/IconButton.tsx

```diff
@@ -234,31 +247,60 @@ const getIconStyle = (size: Size) => css`
  * @param props.variant Determines color of `IconButton`. Can be `light` or `dark`.
  * @param props.href Destination URL, if supplied `IconButton` will render in `a` tags, rather than `button` tags.
  * @param props.onClick Callback fired when `IconButton` is clicked.
- * @param props.ariaLabel Required prop that will be passed to `aria-label` attribute
  * @param props.active Determines whether `IconButton` will appear `active`
  *
  */
 
 const IconButton = React.forwardRef((props: IconButtonProps, ref) => {
   const {
-    variant = 'light',
+    variant = Variant.Light,
+    size = Size.Default,
     disabled = false,
-    size = 'default',
     active = false,
     className,
     href,
     children,
-    ariaLabel,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
     ...rest
   } = props;
 
-  const renderIconButton = (Root: React.ElementType<any> = 'button') => (
+  // We do our own proptype validation here to ensure either 'aria-label' or 'aria-labelledby' are passed to the component.
+  if (!ariaLabel && !ariaLabelledBy) {
+    console.error(
+      'For screen-reader accessibility, aria-label or aria-labelledby must be provided to IconButton.',
+    );
+  }
+
+  const processedChildren = React.Children.map(children, child => {
+    if (isComponentType(child, 'Icon')) {
+      const { size: childSize, title }: IconProps = child.props;
+
+      const newChildProps: Partial<IconProps> = {
+        size: childSize || size,
+      };
+
+      if (typeof title !== 'string' || title.length === 0) {
+        // Unsets the title within an icon since the button itself will have
+        // aria-label or aria-labelledby set.
+        newChildProps.title = false;
+      }
+
+      return React.cloneElement(child, newChildProps);
+    }
+
+    return child;
+  });
+
+  const renderIconButton = (Root: React.ElementType = 'button') => (
     <Root
       {...rest}
-      href={href ? href : undefined}
+      ref={ref}
+      href={href ?? undefined}
```

Comment:
Is this just so it's not null?

Reply:
Good point lol

I updated this without really thinking why we don't just pass href in directly.